sympy/solvers/solveset.py

```diff
@@ -1175,7 +1175,17 @@ def _invert_modular(modterm, rhs, n, symbol):
         if g is not S.One:
             x_indep_term = rhs*invert(g, m)
             return _invert_modular(Mod(h, m), Mod(x_indep_term, m), n, symbol)
-
+    if a.is_polynomial():
```

Comment:
@abhinav28071999 The main problem with this PR is that you code added in `_invert_modular` does not fulfill the goal of `_invert_modular`.
Let me explain it briefly.
`_invert_modular` is supposed to return the inverted form of the equation so that the solution of inverted form should give all solutions of the original equation. And here it is missing.